syncto/tests/test_functional.py

```diff
@@ -0,0 +1,153 @@
+            "Provide a BID assertion %s header." % AUTHORIZATION_HEADER)
+
+    def test_client_state_header_is_required_for_collection(self):
+        headers = self.headers.copy()
+        headers['Authorization'] = "BrowserID abcd"
+        resp = self.app.get(
+            self.collection_url.format(collection_id="tabs"),
+            headers=headers, status=401)
+
+        self.assertFormattedError(
+            resp, 401, ERRORS.MISSING_AUTH_TOKEN, "Unauthorized",
+            "Provide the tokenserver %s header." % CLIENT_STATE_HEADER)
+
+    def test_authorization_header_is_required_for_records(self):
+        resp = self.app.get(
+            self.record_url.format(collection_id="tabs"),
+            headers=self.headers, status=401)
+
+        self.assertFormattedError(
+            resp, 401, ERRORS.MISSING_AUTH_TOKEN, "Unauthorized",
+            "Provide a BID assertion %s header." % AUTHORIZATION_HEADER)
+
+    def test_client_state_header_is_required_for_records(self):
+        headers = self.headers.copy()
+        headers['Authorization'] = "BrowserID abcd"
+        resp = self.app.get(
+            self.record_url.format(collection_id="tabs"),
+            headers=headers, status=401)
+
+        self.assertFormattedError(
+            resp, 401, ERRORS.MISSING_AUTH_TOKEN, "Unauthorized",
+            "Provide the tokenserver %s header." % CLIENT_STATE_HEADER)
+
+
+class CollectionTest(FormattedErrorMixin, BaseWebTest, unittest.TestCase):
+
+    def setUp(self):
+        self.collection_url = COLLECTION_URL.format(collection_id="tabs")
+        self.headers.update({
+            AUTHORIZATION_HEADER: "BrowserID abcd",
+            CLIENT_STATE_HEADER: "1234",
+            'Origin': 'http://localhost:8000'
+        })
+        p = mock.patch("syncto.views.collection.build_sync_client")
+        self.sync_client = p.start()
+        self.sync_client.return_value._authenticate.return_value = None
+        self.sync_client.return_value.get_records.return_value = [{
+            "id": "Y_-5-LEeQBuh60IT0MyWEQ",
+            "modified": 14377478425.69
+        }]
+        headers = {}
+        headers['X-Last-Modified'] = '14377478425.69'
+        headers['X-Weave-Records'] = '1'
+        headers['X-Weave-Next-Offset'] = '12345'
+        self.sync_client.return_value.raw_resp.headers = headers
+
+        self.addCleanup(p.stop)
+
+    def test_collection_handle_cors_headers(self):
+        resp = self.app.get(self.collection_url,
+                            headers=self.headers, status=200)
+        self.assertIn('Access-Control-Allow-Origin', resp.headers)
+
+    def test_collection_handle_since_parameter(self):
+        self.app.get(self.collection_url,
+                     params={'_since': '14377478425700',
+                             '_sort': 'newest'},
+                     headers=self.headers, status=200)
+        # XXX: We should check that the sync_client is called with the
```

Comment:
nit: in order to remember that this test is yet to implement, you can decorate with ``@unittest.skip("todo test sync client")``